api/src/main/java/edu/wpi/first/shuffleboard/api/sources/SourceType.java

```diff
@@ -1,38 +1,40 @@
-package edu.wpi.first.shuffleboard.sources;
-
-import edu.wpi.first.shuffleboard.DummySource;
-import edu.wpi.first.shuffleboard.data.DataType;
+package edu.wpi.first.shuffleboard.api.sources;
 
 import java.util.function.Function;
-import java.util.stream.Stream;
-
-public enum SourceType {
 
-  NONE(false, "", null),
-  STATIC(false, "example://", s -> DummySource.forTypes(DataType.forName(s)).get()),
-  NETWORK_TABLE(true, "network_table://", NetworkTableSource::forKey),
-  CAMERA_SERVER(true, "camera_server://", __ -> {
-    throw new UnsupportedOperationException("Not implemented");
-  });
+public class SourceType {
 
-  public final boolean isRecordable;
+  private final String name;
+  private final boolean isRecordable;
   private final String protocol;
   private final Function<String, DataSource> sourceSupplier;
```

Comment:
IMO it's nice to b able to just call the constructor and pass it the function it needs, rather than making it an abstract method and forcing anonymous classes or a subclass for each type